fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/simplification/CompensateRecordConstructorRule.java

```diff
@@ -59,28 +59,33 @@ public CompensateRecordConstructorRule() {
     public void onMatch(@Nonnull final ValueComputationRuleCall<Iterable<? extends Value>, Map<Value, Function<Value, Value>>> call) {
         final var bindings = call.getBindings();
         final var recordConstructorValue = bindings.get(rootMatcher);
-        final var mergedMatchedValuesMap = new LinkedIdentityMap<Value, Function<Value, Value>>();
+        final var resultingMatchedValuesMap = new LinkedIdentityMap<Value, Function<Value, Value>>();
 
-        for (int i = 0; i < recordConstructorValue.getColumns().size(); ++i) {
-            final var column = recordConstructorValue.getColumns().get(i);
-            final var childResultPair = call.getResult(column.getValue());
-            if (childResultPair == null) {
-                continue;
-            }
+        final var recordConstructorValueResult = call.getResult(recordConstructorValue);
+        if (recordConstructorValueResult != null) {
+            resultingMatchedValuesMap.put(recordConstructorValue, recordConstructorValueResult.getValue().get(recordConstructorValue));
```

Comment:
Can we avoid the extra `Map` by doing something like:

```java
call.yield(recordConstructorValue, recordConstructorValueResult.getRight());
return;
```

Reply:
There may be other values in that map that I don't want to return.